src/hypertable.c

```diff
@@ -2029,26 +1980,196 @@ ts_hypertable_create_internal(PG_FUNCTION_ARGS, bool is_dist_call)
+	Oid sizing_func = PG_ARGISNULL(12) ? InvalidOid : PG_GETARG_OID(12);
+	regproc open_partitioning_func = PG_ARGISNULL(13) ? InvalidOid : PG_GETARG_OID(13);
+	bool replication_factor_is_null = PG_ARGISNULL(14);
+	int32 replication_factor_in = replication_factor_is_null ? 0 : PG_GETARG_INT32(14);
+	ArrayType *data_node_arr = PG_ARGISNULL(15) ? NULL : PG_GETARG_ARRAYTYPE_P(15);
+
+	bool distributed_is_null;
+	bool distributed;
+
+	if (!OidIsValid(table_relid))
+		ereport(ERROR,
+				(errcode(ERRCODE_INVALID_PARAMETER_VALUE), errmsg("relation cannot be NULL")));
+
+	if (!open_dim_name)
+		ereport(ERROR,
+				(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+				 errmsg("partition column cannot be NULL")));
+
+	/* create_distributed_hypertable() does not have explicit
+	 * distributed argument */
+	if (!is_dist_call)
+	{
+		distributed_is_null = PG_ARGISNULL(16);
+		distributed = distributed_is_null ? false : PG_GETARG_BOOL(16);
+	}
+	else
+	{
+		distributed_is_null = false;
+		distributed = true;
+	}
+
+	DimensionInfo *open_dim_info =
+		ts_dimension_info_create_open(table_relid,
+									  open_dim_name,		 /* column name */
+									  default_interval,		 /* interval */
+									  interval_type,		 /* interval type */
+									  open_partitioning_func /* partitioning func */
+		);
+
+	DimensionInfo *closed_dim_info =
+		closed_dim_name == NULL ?
+			NULL :
+			ts_dimension_info_create_closed(table_relid,
+											closed_dim_name,		 /* column name */
+											num_partitions,			 /* number partitions */
+											closed_partitioning_func /* partitioning func */
+			);
+
+	return ts_hypertable_create_internal(fcinfo,
+										 table_relid,
+										 open_dim_info,
+										 closed_dim_info,
+										 associated_schema_name,
+										 associated_table_prefix,
+										 create_default_indexes,
+										 if_not_exists,
+										 migrate_data,
+										 target_size,
+										 sizing_func,
+										 replication_factor_is_null,
+										 replication_factor_in,
+										 data_node_arr,
+										 distributed_is_null,
+										 distributed,
+										 false);
+}
+
 Datum
 ts_hypertable_create(PG_FUNCTION_ARGS)
 {
-	return ts_hypertable_create_internal(fcinfo, false);
+	return ts_hypertable_create_time_prev(fcinfo, false);
 }
 
 Datum
 ts_hypertable_distributed_create(PG_FUNCTION_ARGS)
 {
-	return ts_hypertable_create_internal(fcinfo, true);
+	return ts_hypertable_create_time_prev(fcinfo, true);
+}
+
+static Oid
+get_sizing_func_oid()
+{
+	const char *sizing_func_name = "calculate_chunk_interval";
+	const int sizing_func_nargs = 3;
+	static Oid sizing_func_arg_types[] = { INT4OID, INT8OID, INT8OID };
+
+	return ts_get_function_oid(sizing_func_name,
+							   INTERNAL_SCHEMA_NAME,
+							   sizing_func_nargs,
+							   sizing_func_arg_types);
+}
+
+/*
+ * Process create_hypertable parameters for generic implementation.
+ *
+ * Arguments:
+ * relation                REGCLASS
+ * partition_column        NAME
```

Comment:
again arguments in comments not matching actual used in the function below

Reply:
Fixed.